service-config/src/main/java/com/palantir/remoting2/config/service/ProxyConfiguration.java

```diff
@@ -36,45 +36,105 @@
 @Style(visibility = Style.ImplementationVisibility.PACKAGE, builder = "new")
 public abstract class ProxyConfiguration {
```

Comment:
please update the ServiceConfiguration#proxyConfiguration docs: if absent, uses system proxy configuration.